### PR TITLE
net: add all combinations of wait methods for link/config up/down.

### PR DESCRIPTION
**Pull request by Dirbaio**
- **net-esp-hosted: remove useless fn init.**
- **net-esp-hosted: set wpa3_supported=true.**
- **net: add all combinations of wait methods for link/config up/down.**

### CODE REVIEW

File: embassy-net-esp-hosted/src/control.rs [hunk 1]
```diff
@@ -120,7 +120,7 @@ impl<'a> Control<'a> {
             pwd: unwrap!(String::try_from(password)),
             bssid: String::new(),
             listen_interval: 3,
-            is_wpa3_supported: false,
+            is_wpa3_supported: true,
```
Comment (plaes) [inline embassy-net-esp-hosted/src/control.rs:123]: This commit message should be as TODO in code..

Comment (Dirbaio) [reply embassy-net-esp-hosted/src/control.rs:123]: what's todo? WPA3 works flawlessly now.

Comment (plaes) [reply embassy-net-esp-hosted/src/control.rs:123]: Commit came with following message:
> I've noticed wpa3 still works without this flag, so I'm not sure what this does tbh...

Comment (Dirbaio) [reply embassy-net-esp-hosted/src/control.rs:123]: if WPA3 is working well, the chances of that TODO ever getting looked at are basically zero... I don't think adding it is worth it.

**Pull request merged**